iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/LongRunInstructionCheck.java

```diff
@@ -0,0 +1,136 @@
+    }
+  }
+
+  private List<TooLongLine> computeTooLongLines(RunInstruction runInstruction) {
+    List<TooLongLine> result = new ArrayList<>();
+    var runInstructionData = computeRunInstructionDataPerLines(runInstruction);
+
+    for (Map.Entry<Integer, Integer> linesWithOffsets : runInstructionData.tooLongLinesWithLastOffset.entrySet()) {
+      int line = linesWithOffsets.getKey();
+      int startOffset = runInstructionData.firstOffsetPerLine.get(line);
+      int endOffset = linesWithOffsets.getValue();
+      if (runInstructionData.wordsPerLine.getOrDefault(line, 0) >= minWordsToTrigger) {
+        result.add(new TooLongLine(line, startOffset, endOffset));
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Gather data about a {@link RunInstruction}:
+   * - the first offset of each line, required to raise issue precisely
+   * - the too long lines number with the last offset exceeding the {@link #maxLength} value
+   * - the number of words per lines, to not raise an issue when below the {@link #minWordsToTrigger} value
+   */
+  private RunInstructionData computeRunInstructionDataPerLines(RunInstruction runInstruction) {
+    Map<Integer, Integer> wordsPerLine = countWordsPerLine(runInstruction);
+    var runInstructionData = new RunInstructionData(new HashMap<>(), new HashMap<>(), wordsPerLine);
+
+    TreeVisitor<TreeContext> visitor = new TreeVisitor<>();
+    visitor.register(SyntaxToken.class, (TreeContext ctx, SyntaxToken token) -> {
+      runInstructionData.firstOffsetPerLine.computeIfAbsent(token.textRange().start().line(), line -> token.textRange().start().lineOffset());
+      // Special case of token that finish on a different line: consider the first offset of the end line to start at 0.
+      if (token.textRange().start().line() != token.textRange().end().line()) {
+        runInstructionData.firstOffsetPerLine.put(token.textRange().end().line(), 0);
+      }
+      if (token.textRange().end().lineOffset() > maxLength) {
+        runInstructionData.tooLongLinesWithLastOffset.put(token.textRange().end().line(), token.textRange().end().lineOffset());
+      }
+    });
+    visitor.scan(new TreeContext(), runInstruction);
+
+    return runInstructionData;
+  }
+
+  private static Map<Integer, Integer> countWordsPerLine(RunInstruction runInstruction) {
+    Map<Integer, Integer> wordsPerLine = new HashMap<>();
+    incrementLine(wordsPerLine, runInstruction.keyword());
+    runInstruction.options().forEach(flag -> incrementLine(wordsPerLine, flag));
+    runInstruction.arguments().forEach(argument -> incrementLine(wordsPerLine, argument));
+    return wordsPerLine;
+  }
+
+  private static void incrementLine(Map<Integer, Integer> wordsPerLine, Tree tree) {
+    if (tree.textRange().start().line() != tree.textRange().end().line()) {
+      incrementLine(wordsPerLine, tree.textRange().start().line());
+    }
+    incrementLine(wordsPerLine, tree.textRange().end().line());
```

Comment:
Isn't this essentially the same as always using `start()`?

Reply:
Not sure what you mean here.

The purpose is to increase the counter of lines on which the tree is present.
But, we should not increase the counter twice if the start line and end line are the same.
So here I'm increasing the end line every time, and the start line only if it's different.